src/symbolicator.index.js

```diff
@@ -0,0 +1,150 @@
+        breakpadId: string
+      ): Promise<SymbolTableAsTuple> {
+        return new Promise((resolve, reject) => {
+            const key = this._makeKey(debugName, breakpadId);
+            if (key in this._dict) {
+                resolve(this._dict[key]);
+            } else {
+                reject(new SymbolsNotFoundError(
+                    'The requested library does not exist in the database.',
+                    { debugName, breakpadId }
+                  ));
+            }
+        });
+      }
+
+    close(): Promise<void> { return new Promise((resolve, reject) => resolve()); }
+};
+
+const symbolStoreDB = new InMemorySymbolDB();
+
+/**
+ * SymbolStore implementation which just forwards everything to the symbol server in
+ * MozillaSymbolicationAPI format. No support for getting symbols from 'the browser' as
+ * there is no browser in this context.
+ */
+const symbolStore = new SymbolStore(symbolStoreDB, {
+    requestSymbolsFromServer: async (requests) => {
+      for (const { lib } of requests) {
+        console.log(`  Loading symbols for ${lib.debugName}`);
+      }
+      try {
+        return await MozillaSymbolicationAPI.requestSymbols(
+          'symbol server',
+          requests,
+          async (path, json) => {
+            const response = await fetch(argv.server + path, {
+              body: json,
+              method: 'POST'
+            });
+            return response.json();
+          }
+        );
+      } catch (e) {
+        throw new Error(
+          `There was a problem with the symbolication API request to the symbol server: ${e.message}`
+        );
+      }
+    },
+
+    requestSymbolsFromBrowser: async (requests) => {
+        return [];
+    },
+
+    requestSymbolTableFromBrowser: async (lib) => {
+        return [];
+    },
+  });
+
+console.log("Symbolicating...");
+
+const symbolicationStepsPerThread : Map<ThreadIndex, SymbolicationStepInfo[]> = new Map();
+await symbolicateProfile(
+    profile,
+    symbolStore,
+    (
+        threadIndex: ThreadIndex,
+        symbolicationStepInfo: SymbolicationStepInfo
+      ) => {
+        let threadSteps = symbolicationStepsPerThread.get(threadIndex);
+        if (threadSteps === undefined) {
+          threadSteps = [];
+          symbolicationStepsPerThread.set(threadIndex, threadSteps);
+        }
+        threadSteps.push(symbolicationStepInfo);
+      }
+);
+
+console.log("Applying collected symbolication steps...");
+
+profile.threads = profile.threads.map((oldThread, threadIndex) => {
+  const symbolicationSteps = symbolicationStepsPerThread.get(threadIndex);
+  if (symbolicationSteps === undefined) {
+    return oldThread;
+  }
+  const { thread, _ } = applySymbolicationSteps(oldThread, symbolicationSteps);
+  return thread;
+});
+
+profile.meta.symbolicated = true;
+
+console.log(`Saving profile to ${argv.output}`);
+fs.writeFileSync(argv.output, JSON.stringify(profile));
+console.log("Finished.");
+process.exit(0);
```

Comment:
I think you don't need to call this exit explicitely, right?

Reply:
Possibly not; I felt it was better to be explicit about the exit code but I can remove it if you prefer.

Reply:
The problem with `process.exit` is that you basically tell node to exit right away, without waiting for any extra operations that might still be happening. This might not be a problem for this specific program in its current state, but I'd rather remove it.